### PR TITLE
Fiorina Riot In Progress: Colonial Marshals & United Americas Police

### PR DESCRIPTION
**Pull request by QuickLode**
# About the pull request

Howdy, this PR adds in unique Riot Response flavored CMB Deputies and UA Riot Police to a Survivor Nightmare(thx badatthisgame302) on Fiorina Science Annex. Because this is a survivor nightmare I kept the loadouts very sparse, so you will only find a couple mags of ammo per weapon, and all participants here will likely have to resort to their sidearm. It is implied that by the time players control these characters, they've been surviving for hours if not days. Furthermore, because its a riot response, there are no specialists (engineers, doctors, etc.) to help the support side of things. After all, it is just a riot call..

PS. If you haven't read the Alien: Isolation comic yet, something similar is implied there. :) 

On the CMB side we have 3 variations and a Synthetic. These investigative federal agents are nimble on their feet, but inadequately protected and their weapons are more focused on self defense. They were sent here as emergency back-up to the understaffed prison, and aren't prepared for this scenario. They do however still carry some rubber munitions and crowd control elements.

On the UA side we have 4 variations and a Synthetic. The infamous United Americas Riot Control Police show with four variations. Donned in complete riot gear and with uniformity, they would seem like an unstoppable force to be reckoned with. Unfortunately for them, they're dealing with Xenomorphs - all of their crowd control devices and nonlethal munitions may prove to be very ineffective at dealing with this new threat at hand..

# Explain why it's good for the game

Adding meaningful and fitting equipment really adds to the immersion. Its not just a setting but it envokes the real feeling of "We came here to stop a riot, but we found something much more sinister." By giving players these settings to play with, they'll feel much more at home in the Aliens universe as opposed to the looming feeling of it being "just another round, just another role."

I think CM really benefits from spicing up the loop and nightmares are an excellent example of that. Having lore accurate, flavorful layouts like this where you have a consistent theme going allows for excellent immersion. Even if everyone dies, the Marines stumbling upon the corpses of a horrific incident like this will be a really unique experience, in my opinion.

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

Basically I created a new file for these fine people and added what I think is thematically fitting for this scenario. I then tested it, fixed bugs, and then tested it again. This happened 3 times until I could no longer find any bugs. 

https://cdn.discordapp.com/attachments/1042176396711170119/1222482124771430460/image.png?ex=66166033&is=6603eb33&hm=69719c33b5d811033f62cba605d5ba6e044c4cdd8f7056c50e49e9523e7ce3f8&
</details>

# Changelog

:cl:
add: Adds 3 variations of CMB Marshal and 4 variations of a UA Riot Police Officer to Fiorina Prison Riot nightmare. These are fine men and women in uniform prepared to engage what they suspect is a riot(or in the case of CMB, assisting their underfunded and understaffed colleagues) however, they find something much more sinister going on..
add: Adds a CMB Investigative Synthetic(r) and a UA Police Synthetic.
add: Allows police baton and telescopic baton to shield clash for intimidation and riot tactics.
add: Allows M81 launcher to utilize less lethal baton round
qol: removes taser from some security synthetics belts, replaced with a more synthetic friendly version
/:cl:

### CODE REVIEW

**Review by SabreML — CHANGES_REQUESTED** [review]
Sorry for the delay!
Just a couple of things at the moment. I'll look through the new equipment presets more thoroughly once the pathing review has been resolved. 👍

File: code/modules/gear_presets/survivors/fiorina_sciannex/riot_in_progress_insert_fiorina_nightmare.dm [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+
+/datum/equipment_preset/synth/survivor/cmb/synth/load_race(mob/living/carbon/human/new_human)
+	new_human.set_species(SYNTH_COLONY)
+
+/datum/equipment_preset/synth/survivor/cmb/synth/load_gear(mob/living/carbon/human/new_human)
+	//backpack
+	new_human.equip_to_slot_or_del(new /obj/item/storage/backpack/security, WEAR_BACK)
+	new_human.equip_to_slot_or_del(new /obj/item/ammo_magazine/revolver/cmb/normalpoint, WEAR_IN_BACK)
+	new_human.equip_to_slot_or_del(new /obj/item/ammo_magazine/revolver/cmb/normalpoint, WEAR_IN_BACK)
+	new_human.equip_to_slot_or_del(new /obj/item/ammo_magazine/revolver/cmb, WEAR_IN_BACK)
```
Comment (Drulikar) [inline code/modules/gear_presets/survivors/fiorina_sciannex/riot_in_progress_insert_fiorina_nightmare.dm:106]: Why are they getting mags for a pistol they don't have nor can fire?

File: code/modules/gear_presets/survivors/fiorina_sciannex/riot_in_progress_insert_fiorina_nightmare.dm [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+	new_human.equip_to_slot_or_del(new /obj/item/handcuffs/zip, WEAR_IN_ACCESSORY)
+	new_human.equip_to_slot_or_del(new /obj/item/handcuffs/zip, WEAR_IN_ACCESSORY)
+	new_human.equip_to_slot_or_del(new /obj/item/handcuffs/zip, WEAR_IN_ACCESSORY)
+	new_human.equip_to_slot_or_del(new /obj/item/storage/box/flashbangs, WEAR_IN_ACCESSORY)
+	new_human.equip_to_slot_or_del(new /obj/item/explosive/grenade/custom/teargas, WEAR_IN_ACCESSORY)
+	new_human.equip_to_slot_or_del(new /obj/item/clothing/suit/storage/marine/veteran/ua_riot/synth, WEAR_JACKET)
+	new_human.equip_to_slot_or_del(new /obj/item/weapon/telebaton, WEAR_IN_JACKET)
+	new_human.equip_to_slot_or_del(new /obj/item/explosive/grenade/flashbang, WEAR_IN_JACKET)
+	//belt
+	new_human.equip_to_slot_or_del(new /obj/item/storage/belt/security/MP/full, WEAR_WAIST)
```
Comment (Drulikar) [inline code/modules/gear_presets/survivors/fiorina_sciannex/riot_in_progress_insert_fiorina_nightmare.dm:268]: This contains a `/obj/item/weapon/gun/energy/taser` they cannot fire.

File: code/modules/gear_presets/survivors/fiorina_sciannex/riot_in_progress_insert_fiorina_nightmare.dm [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+	new_human.set_species(SYNTH_COLONY)
+
+/datum/equipment_preset/synth/survivor/cmb/ua_synth/load_gear(mob/living/carbon/human/new_human)
+	//backpack
+	new_human.equip_to_slot_or_del(new /obj/item/storage/backpack/satchel/sec, WEAR_BACK)
+	new_human.equip_to_slot_or_del(new /obj/item/handcuffs/zip, WEAR_IN_BACK)
+	new_human.equip_to_slot_or_del(new /obj/item/handcuffs/zip, WEAR_IN_BACK)
+	new_human.equip_to_slot_or_del(new /obj/item/explosive/grenade/custom/teargas, WEAR_IN_BACK)
+	new_human.equip_to_slot_or_del(new /obj/item/explosive/grenade/custom/teargas, WEAR_IN_BACK)
+	new_human.equip_to_slot_or_del(new /obj/item/storage/box/packet/baton_slug, WEAR_IN_BACK)
```
Comment (Drulikar) [inline code/modules/gear_presets/survivors/fiorina_sciannex/riot_in_progress_insert_fiorina_nightmare.dm:247]: What are they going to fire slugs from? They cannot fire a `/obj/item/weapon/gun/launcher/grenade/m81/riot`

File: code/modules/gear_presets/survivors/fiorina_sciannex/riot_in_progress_insert_fiorina_nightmare.dm [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+	new_human.equip_to_slot_or_del(new /obj/item/ammo_magazine/rifle/m16, WEAR_IN_BACK)
```
Comment (Drulikar) [inline code/modules/gear_presets/survivors/fiorina_sciannex/riot_in_progress_insert_fiorina_nightmare.dm:248]: What m16 can they fire?

**Review by Drulikar — CHANGES_REQUESTED** [review]

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

File: code/game/objects/items/storage/belt.dm [hunk 5]
```diff
@@ -383,7 +383,13 @@
...
 	new /obj/item/device/clue_scanner(src)
 
-
+/obj/item/storage/belt/security/MP/full/synth/fill_preset_inventory()
+	new /obj/item/explosive/grenade/flashbang(src)
+	new /obj/item/device/flash(src)
+	new /obj/item/weapon/baton(src)
+	new /obj/item/reagent_container/spray/pepper(src)
+	new /obj/item/device/clue_scanner(src)
+	new /obj/item/handcuffs(src)
```
Comment (Drulikar) [inline code/game/objects/items/storage/belt.dm:392]:
```suggestion
	new /obj/item/handcuffs(src)

```

File: code/modules/clothing/suits/marine_armor/ert.dm [hunk 6]
```diff
@@ -731,6 +731,20 @@
...
 	flags_atom = NO_SNOW_TYPE
 
+/obj/item/clothing/suit/storage/marine/veteran/ua_riot/synth
+	name = "\improper UA-M1S Synthetic body armor"
+	desc = "Based on the M-3 pattern employed by the USCM, the UA-M1 body armor is employed by UA security, riot control and union-busting teams. The UA-1MS modification is Synthetic programming compliant, sacrificing protection for speed and carrying capacity."
+	armor_melee = CLOTHING_ARMOR_NONE
+	armor_bullet = CLOTHING_ARMOR_NONE
+	armor_energy = CLOTHING_ARMOR_NONE
+	armor_bomb = CLOTHING_ARMOR_NONE
+	armor_rad = CLOTHING_ARMOR_NONE
```
Comment (Drulikar) [inline code/modules/clothing/suits/marine_armor/ert.dm:741]:
```suggestion
	armor_bullet = CLOTHING_ARMOR_NONE
	armor_laser = CLOTHING_ARMOR_NONE
	armor_energy = CLOTHING_ARMOR_NONE
	armor_bomb = CLOTHING_ARMOR_NONE
	armor_bio = CLOTHING_ARMOR_NONE
	armor_rad = CLOTHING_ARMOR_NONE
```
Missing value resets for laser and bio.

**Review by Drulikar — CHANGES_REQUESTED** [review]

**Review by Drulikar — APPROVED** [review]

**Comment by Skatuk** [comment]
I'm sorry but this doesn't look good on paper, not a single engineer or medic and all depended on a synthetic to keep them alive. What if that synthetic doesn't spawn since its a WL only? Also once T2's spawn its basically over since they have no cades to be protected by.

**Pull request merged**